Portal_Perfomance_Employes/Controllers/UserEvaluationController.cs

```diff
@@ -0,0 +1,139 @@
+            if (empSkills != null)
+            {
+                var hardSkillsTable = new DataTable();
+                hardSkillsTable.Columns.Add("SkillName", typeof(string));
+                hardSkillsTable.Columns.Add("SkillLevel", typeof(string));
+
+                foreach (var empSkill in empSkills)
+                {
+                    var skill = await _context.Skills.FirstOrDefaultAsync(s => s.SkillId == empSkill.SkillId);
+                    var skillType = await _context.SkillTypes.FirstOrDefaultAsync(st => st.SkillTypeId == skill.SkillTypeId);
+                    var skillLevel = await _context.SkillLevels.FirstOrDefaultAsync(sl => sl.SkillLevelId == empSkill.SkillLevelId);
+
+                    if (skillType.SkillTypeName == "Soft skill")
+                    {
+                        hardSkillsTable.Rows.Add(
+                            skill.SkillName,
+                            skillLevel.SkillLevelName
+                        );
+                    }
+                }
+                return Ok(hardSkillsTable.ToJson());
+            }
+            return NotFound("The user skills list is empty");
+        }
+
+        [Route("GetUserHardSkills")]
+        [HttpGet]
+        public async Task<IActionResult> GetUserHardSkills()
+        {
+            int.TryParse(User.FindFirst(ClaimTypes.NameIdentifier)?.Value, out int id);
+            var empSkills = await _context.EmployeeSkills.Where(e => e.EmployeeId == id).ToListAsync();
+            if (empSkills != null)
+            {
+                var hardSkillsTable = new DataTable();
+                hardSkillsTable.Columns.Add("SkillName", typeof(string));
+                hardSkillsTable.Columns.Add("SkillLevel", typeof(string));
+
+                foreach (var empSkill in empSkills)
+                {
+                    var skill = await _context.Skills.FirstOrDefaultAsync(s => s.SkillId == empSkill.SkillId);
+                    var skillType = await _context.SkillTypes.FirstOrDefaultAsync(st => st.SkillTypeId == skill.SkillTypeId);
+                    var skillLevel = await _context.SkillLevels.FirstOrDefaultAsync(sl => sl.SkillLevelId == empSkill.SkillLevelId);
+
+                    if (skillType.SkillTypeName == "Hard skill")
+                    {
+                        hardSkillsTable.Rows.Add(
+                            skill.SkillName,
+                            skillLevel.SkillLevelName
+                        );
+                    }
+                }
+                return Ok(hardSkillsTable.ToJson());
+            }
+            return NotFound("The user skills list is empty");
+        }
+
+        [Route("AddUserSkill")]
+        [HttpPut]
+        public async Task<IActionResult> AddUserSkill(int skillType, string skillName, int skillLevelId, int skillLevelTypeId)
+        {
+            int.TryParse(User.FindFirst(ClaimTypes.NameIdentifier)?.Value, out int id);

```

Comment:
if fails to parse and 0 by default? could cause some issues if we have something with id 0 in db

Reply:
Isn't FindFirst return null if match not found?